kimjaegwan/src/pages/index.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { ChangeEvent, FormEvent, useState } from 'react';
+import styled from 'styled-components';
+import instance from '../libs/instances';
+import useTodos from '../hooks/useTodos';
+import { config } from 'process';
+const Home = () => {
+    const { todos, saveTodos } = useTodos();
+    const [newTodo, setNewTodo] = useState<string>('');
+
+    const _onChange = (e: ChangeEvent<HTMLInputElement>) => {
+        const { target } = e;
+        const { value } = target;
+        setNewTodo(value);
+    };
+
+    const _onSubmit = (e: FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+
+        instance //이것도 어찌저찌하면 hooks안에 넣을 수 있을거같다
+            .post('/todos', {
+                records: [
+                    {
+                        fields: {
+                            Name: newTodo,
+                        },
+                    },
+                ],
+            })
+            .then(() => {
+                saveTodos();
+                // ? setNewTodo(''); 왜 여기에 들어가면 작동을 하지 않는지 이해가가지 않음
+            });
+        setNewTodo('');
+    };
+
+    //삭제 기능 구현
+    const _onDelete = (e: React.MouseEvent<HTMLButtonElement>) => {
+        console.log(e.currentTarget.id);
+        instance.delete(`/todos/${e.currentTarget.id}`).then(() => {
+            saveTodos();
+        });
+        setNewTodo('');
+    };
+
+    //수정 기능 구현
+    const _onUpdate = (e: React.MouseEvent<HTMLButtonElement>) => {
+        e.preventDefault();
+        if (newTodo === '') {
+            alert('내용을 입력하고 수정하세요.');
+            return;
+        }
+
+        instance
+            .patch('/todos', {
+                records: [
+                    {
+                        id: e.currentTarget.id,
+                        fields: {
+                            Name: newTodo,
+                        },
+                    },
+                ],
+            })
+            .then(() => {
+                saveTodos();
+                // ? setNewTodo(''); 왜 여기에 들어가면 작동을 하지 않는지 이해가가지 않음
+            });
+        setNewTodo('');
+    };
+
+    return (
+        <>
+            <br />
+            <h1>TodoList</h1>
+            <StyledHomeBox>
+                {todos.map((data) => (
+                    <StyledTodo key={data.id}>
+                        <p key={data.id}>{data.fields.Name} </p>
+                        <div>
+                            <button onClick={_onDelete} id={data.id}>
```

Comment:
이렇게 버튼에 id를 넣는것도 새로 알아갑니다🙃👍

Reply:
권장되는 방법인진 모르겠으나 빼올 때 편했습니다!

Reply:
DOM ID를 사용하는 것도 방법이지만,

저는 개인적으로 느슨하게 연결되어 있다고 생각해서 사용하는 편은 아니에요

자바스크립트의 커링을 이용해서 이렇게 표현하는 걸 선호하는 편 이에요

```suggestion
const _onDelete = (id: number) => (event: MouseEvent<HTMLButtonElement>) => {
// logic
}

                            <button onClick={_onDelete(data.id)}>
```